exchanges/subscription/template.go

```diff
@@ -69,11 +71,26 @@ func (l List) ExpandTemplates(e IExchange) (List, error) {
 
 	subs := List{}
 	for _, s := range l {
+		// Validate the subscription before expansion to capture fields that may not be used in the template
+		if v, ok := e.(SubValidator); ok {
+			if err := v.ValidateSubscription(s); err != nil {
+				return nil, fmt.Errorf("validate subscription: %w", err)
+			}
+		}
+
 		expanded, err2 := expandTemplate(e, s, maps.Clone(ap), assets)
 		if err2 != nil {
 			err = common.AppendError(err, fmt.Errorf("%s: %w", s, err2))
-		} else {
-			subs = append(subs, expanded...)
+			continue
+		}
+
+		subs = append(subs, expanded...)
+	}
+
+	// Validate the subscriptions after expansion to capture fields that will be used in the template
+	if v, ok := e.(ListValidator); ok {
+		if err := v.ValidateSubscriptions(subs); err != nil {
```

Comment:
🚧 
This is only validating the ExpandedTemplates here.
Wouldn't it need to be done at a higher level, when we're subscribing, to also check existing subs that are already subscribed?

Reply:
I don't think we have a good place right now.
We're lacking a cross exchange entrypoint into `Subscribe`, since exchanges have their own implementation of it.
For now, I think the solution is for GateIO's `ValidateSubscriptions` to recast `v` back into `IExchange` to get at the rest of the subscriptions as well as the new `subs`.
That's sub-optimal, but fixes the issue.

I'll bear this problem in mind in #1764

Reply:
Sounds good will give it a crack! 👍

Reply:
Addressed I hope 🙏: https://github.com/thrasher-corp/gocryptotrader/pull/1863/commits/fc96cd2a38e98650859ee26dcfb7eb9b6402d718

Reply:
I think for now this solution is good.
Overall I don't think it should be in ExpandTemplates at all, but we lack a unified subscription "engine" right now above exchange level.
I'll be adding that soon.